868 improving README with AWS API gateway findings (#869)
grpc-ecosystem/grpc-gateway: docs/_docs/aws.md

---
category: documentation
---

# AWS

## Import swagger documentation into AWS API Gateway

> AWS Api gateway service allows to import swagger documentation to create REST API. The process is very straigh foward and can be found [here](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-import-api.html).

Reviewer: ```suggestion
The AWS API gateway service allows importing of a OpenAPI specification to create a REST API. The process is very straightforward and can be found [here](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-import-api.html).
```

Author: ok

> The AWS API gateway service allows importing of a OpenAPI specification to create a REST API. The process is very straightforward and can be found [here](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-import-api.html).

Reviewer: ```suggestion
The AWS API gateway service allows importing of an OpenAPI specification to create a REST API. The process is very straightforward and can be found [here](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-import-api.html).
```

> These are some considerations when importing the documentation:

Reviewer: ```suggestion
Here are some tips to consider when importing the documentation:
```

Author: ok

> 1. Remove model circular dependencies(Circular dependencies aren't supported by the parser).

Reviewer: ```suggestion
1. Remove any circular dependencies (these aren't supported by the parser).
```

Author: ok

2. Remove security-related annotations (These annotations aren't well supported by the parser).

> 3. Max length of fields is reviewed by the parser but the errors aren't self-explanatory. Review the [documentation](https://swagger.io/specification/v2/) to verify that the requirements are met.

Reviewer: ```suggestion
3. Max length of fields are reviewed by the parser but the errors aren't self-explanatory. Review the [documentation](https://swagger.io/specification/v2/) to verify that the requirements are met.
```

Author: ok

> 4. API gateway errors aren't good enough for troubleshooting so I recommend using this [page](https://apidevtools.org/swagger-parser/online/) for structure validation.

Reviewer: ```suggestion
4. API gateway errors aren't great, but you can use this [page](https://apidevtools.org/swagger-parser/online/) for structure validation.
```
